python/jsbeautifier/javascript/beautifier.py

```diff
@@ -848,18 +869,19 @@ def handle_word(self, current_token):
 
         prefix = 'NONE'
 
-        if self._flags.last_token.type == TOKEN.END_BLOCK:
-            if self._previous_flags.inline_frame:
-                prefix = 'SPACE'
-            elif not reserved_array(current_token, ['else', 'catch', 'finally', 'from']):
-                prefix = 'NEWLINE'
-            else:
-                if self._options.brace_style in ['expand', 'end-expand'] or (
-                        self._options.brace_style == 'none' and current_token.newlines):
-                    prefix = 'NEWLINE'
-                else:
-                    prefix = 'SPACE'
-                    self._output.space_before_token = True
+        is_literal_sbrk_cma = self._flags.mode == MODE.ObjectLiteral and self._last_last_text in ['{', ',']
+        token_reserved_word = reserved_array(current_token, ['else', 'catch', 'finally', 'from'])
+        no_brace_style_nl = self._options.brace_style == 'none' and current_token.newlines
+        brace_style_in_exp = self._options.brace_style in ['expand', 'end-expand', 'expand-all']
```

Comment:
New feature to have same behavior as that of brace style option with value expand or end-expand.